File: 0104-maximum-depth-of-binary-tree/0104-maximum-depth-of-binary-tree.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode() : val(0), left(nullptr), right(nullptr) {}
 *     TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
 *     TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
 * };
 */
class Solution {
public:
int fun(TreeNode* root){
    // Agar root hi null hai; yni tree hi nhi hai 
    if(root==nullptr){
        return 0;
    }

    // Left Subtree ka ht
    int left=fun(root->left);
    // Right Subtree ka depth
    int right=fun(root->right);

    // wo node root wla k sath milkar ht hogi
    return 1+max(left,right);
}
    int maxDepth(TreeNode* root) {
        // recursive Function call ka sath hi return kro 
        return fun(root);
    }
};

File: 0104-maximum-depth-of-binary-tree/0104-maximum-depth-of-binary-tree.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode() {}
 *     TreeNode(int val) { this.val = val; }
 *     TreeNode(int val, TreeNode left, TreeNode right) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */
class Solution {
    public int fun(TreeNode root){
        // agar root hai hi nhi yni tree hi nhi hai 
        if(root==null){
            return 0;
        }

        // left Subtree ka Ht
        int left=fun(root.left);
        // right subtree ka hT
        int right=fun(root.right);

        // return karna hai 
        return 1+Math.max(left,right);
    }

    public int maxDepth(TreeNode root) {
        // Recursive fn call k sath return kro 
        return fun(root);
    }
}